service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt

```diff
@@ -0,0 +1,698 @@
+    return res.vardaDecisionId
+}
+
+fun createPlacementToVarda(client: VardaClient, vardaDecisionId: Long, evakaServiceNeedInfoForVarda: EvakaServiceNeedInfoForVarda): Long {
+    val res: VardaPlacementResponse?
+    try {
+        res = client.createPlacement(evakaServiceNeedInfoForVarda.toVardaPlacement(client.getDecisionUrl(vardaDecisionId), client.sourceSystem))
+    } catch (e: Exception) {
+        throw Exception("VardaUpdate: cannot create placement for ${evakaServiceNeedInfoForVarda.id}: varda client threw ${e.localizedMessage}")
+    }
+
+    if (res == null) throw Exception("VardaUpdate: cannot create placement for ${evakaServiceNeedInfoForVarda.id}: create varda placement response is null")
+    return res.vardaPlacementId
+}
+
+fun createFeeDataToVardaFromFeeDecision(
+    db: Database.Connection,
+    client: VardaClient,
+    vardaChildId: Long,
+    evakaServiceNeedInfoForVarda: EvakaServiceNeedInfoForVarda,
+    decision: FeeDecision
+): Long {
+    val guardians = db.read { listOfNotNull(it.getPersonById(decision.headOfFamily.id), if (decision.partner?.id != null) it.getPersonById(decision.partner.id) else null) }
+    if (guardians.isEmpty()) throw Exception("VardaUpdate: could not create fee data for ${evakaServiceNeedInfoForVarda.id}: child has no guardians")
+    val childPart = decision.children.find { part -> part.child.id == evakaServiceNeedInfoForVarda.childId }
+        ?: throw Exception("VardaUpdate: could not create fee data for ${evakaServiceNeedInfoForVarda.id}: fee ${decision.id} has no part for child ${evakaServiceNeedInfoForVarda.childId}")
+
+    val res: VardaFeeDataResponse?
+    try {
+        res = client.createFeeData(
+            VardaFeeData(
+                huoltajat = guardians.map { guardian -> VardaGuardian(guardian.identity.toString(), guardian.firstName ?: "", guardian.lastName ?: "") },
+                lapsi = client.getChildUrl(vardaChildId),
+                maksun_peruste_koodi = vardaFeeBasisByPlacementType(childPart.placement.type),
+                asiakasmaksu = childPart.finalFee.div(100.0),
+                palveluseteli_arvo = 0.0,
+                perheen_koko = decision.familySize,
+                alkamis_pvm = calculateVardaFeeDataStartDate(decision.validFrom, evakaServiceNeedInfoForVarda.asPeriod),
+                paattymis_pvm = minEndDate(evakaServiceNeedInfoForVarda.endDate, decision.validTo),
+                lahdejarjestelma = client.sourceSystem
+            )
+        )
+    } catch (e: Exception) {
+        throw Exception("VardaUpdate: cannot create fee data ${decision.id} for service need ${evakaServiceNeedInfoForVarda.id}: varda client threw ${e.localizedMessage}")
+    }
+
+    if (res == null) throw Exception("VardaUpdate: cannot create fee data ${decision.id} for service need ${evakaServiceNeedInfoForVarda.id}: create varda placement response is null")
```

Comment:
Typo here: `create varda placement response` should be `create varda fee data response`

Reply:
Same applies near row 292